VultisigApp/VultisigApp/Chains/Cardano.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  Cardano.swift
+//  VultisigApp
+//
+//  Created by AI Assistant
+//
+
+import Foundation
+import Tss
+import WalletCore
+import BigInt
+
+enum CardanoHelper {
+    
+    static func getPreSignedInputData(keysignPayload: KeysignPayload) throws -> Data {
+        guard keysignPayload.coin.chain == .cardano else {
+            throw HelperError.runtimeError("coin is not ADA")
+        }
+        
+        guard case .UTXO(let byteFee, let sendMaxAmount) = keysignPayload.chainSpecific else {
+            throw HelperError.runtimeError("fail to get UTXO chain specific byte fee")
+        }
+        
+        guard let toAddress = AnyAddress(string: keysignPayload.toAddress, coin: .cardano) else {
+            throw HelperError.runtimeError("fail to get to address: \(keysignPayload.toAddress)")
+        }
+        
+        // Prevent from accidentally sending all balance
+        var safeGuardMaxAmount = false
+        if let rawBalance = Int64(keysignPayload.coin.rawBalance),
+           sendMaxAmount,
+           rawBalance > 0,
+           rawBalance == Int64(keysignPayload.toAmount) {
+            safeGuardMaxAmount = true
+        }
+        
+        // For Cardano, we don't use UTXOs from Blockchair since it doesn't support Cardano
+        // Instead, we create a simplified input structure
+        var input = CardanoSigningInput.with {
+            $0.transferMessage = CardanoTransfer.with {
+                $0.toAddress = keysignPayload.toAddress
+                $0.changeAddress = keysignPayload.coin.address
+                $0.amount = UInt64(keysignPayload.toAmount)
+                $0.useMaxAmount = safeGuardMaxAmount
+                //$0.forceFee = UInt64(byteFee)
+            }
+            $0.ttl = 1736265600 // Fixed TTL for testing (2025-01-07 16:00:00 UTC) - REPLACE WITH DYNAMIC VALUE LATER
```

Comment:
do it appropriately?

Reply:
We could define a BlockSpecific for that purpose ,  and set TTL to 5 minutes in the future